### PR TITLE
Windows: Statically linked C runtime

### LINKED ISSUE #11503: Docs: Microsoft Visual C++ 2015 Redistributable is required on Windows
**OS**: Windows 10
**ruff**: version 0.4.4 (installed via `pipx`)

Issue
-----
`ruff.exe` silently fails to run from the terminal if Microsoft Visual C++ 2015 Redistributable is not installed.

Running
```console
> ruff --version
```
does not show the version in the terminal nor is an error shown. Complete silence.

Double-clicking the executable, located at `..\pipx\venvs\ruff\Scripts\ruff.exe`, pops-up the following window

vcruntime140-error

Fix
---
Install Microsoft Visual C++ 2015 Redistributable, then

```console
> ruff --version
ruff 0.4.4
```

Suggestion
------------
Perhaps adding something to the docs (https://docs.astral.sh/ruff/installation/) for Windows users that Microsoft Visual C++ 2015 Redistributable must be installed. Either provide the download link (https://www.microsoft.com/en-us/download/details.aspx?id=52685) and/or specify the command to run to install it `winget install --exact --id Microsoft.VCRedist.2015+.x64`

### PR DESCRIPTION
**Pull request by T-256**
## Summary
Here is a comparison between [0.4.6 released binaries](https://github.com/astral-sh/ruff/actions/runs/9274921378) (dynamic-crt) and [current PR binaries](https://github.com/T-256/ruff/actions/runs/9283204908) (static-crt):

| TARGET | dynamic size | static size | increased size |
| - | - | - | - |
| aarch64-pc-windows-msvc | 7.32 MB | 7.36 MB | 0.04 MB |
| i686-pc-windows-msvc | 7.14 MB | 7.2 MB | 0.06 MB |
| x86_64-pc-windows-msvc | 7.85 MB | 7.91 MB | 0.06 MB |

Uncompressed binaries:
| TARGET | dynamic size | static size | increased size |
| - | - | - | - |
| aarch64-pc-windows-msvc | 19.5 MB | 19.6 MB | 0.1 MB |
| i686-pc-windows-msvc | 18.5 MB | 18.6 MB | 0.1 MB |
| x86_64-pc-windows-msvc | 23 MB | 23.1 MB | 0.1 MB |

Closes #11503

### CODE REVIEW

File: .cargo/config.toml [hunk 1]
```diff
@@ -1,3 +1,17 @@
...
+# See: https://github.com/astral-sh/ruff/issues/11503
+[target.x86_64-pc-windows-msvc]
+rustflags = ["-C", "target-feature=+crt-static"]
+[target.i686-pc-windows-msvc]
+rustflags = ["-C", "target-feature=+crt-static"]
+[target.x86_64-unknown-linux-musl]
+rustflags = [
+  "-C", "target-feature=+crt-static",
+  "-C", "link-self-contained=yes",
+]
```
Comment (T-256) [inline .cargo/config.toml:17]: It seems it still does not migrate to dynamic crt, so for now I'll drop it.

```suggestion
```

**Comment by charliermarsh** [comment]
@BurntSushi - tagging you in here too.

**Comment by charliermarsh** [comment]
That size increase looks totally reasonable to me.

**Review by MichaReiser — APPROVED** [review]
Thanks @T-256 for working on this. 

> We also support aarch64 windows target, should I include it in this PR?

Yes, I think it would be good if all windows versions are built the seme way.

**Review by BurntSushi — APPROVED** [review]

**Pull request merged**